File: 034-Sum of Digits is Prime/SumofDigitsisPrime.py
def is_prime(num):
    if num < 2:
        return False
    for i in range(2, int(num**0.5)+1):
        if num % i == 0:
            return False
    return True

# Function to calculate the sum of digits
def sum_of_digits(n):
    return sum(int(digit) for digit in str(n))

# Input and output logic
n = int(input("Enter a Number: "))
digit_sum = sum_of_digits(n)
if is_prime(digit_sum):
    print("YES")
else:
    print("NO")

File: 034-Sum of Digits is Prime/SumofDigitsisPrime.cpp
#include<iostream>
using namespace std;

bool isPrime(int num){
    if(num < 2){
        return false;
    }
    for(int i = 2; i*i <= num; i++){
        if(num % i == 0){
            return false;
        }
    }
    return true;
}

int sumOfDigits(int n){
    int sum = 0;
    while(n > 0){
        sum += n % 10;
        n /= 10;
    }
    return sum;
}

int main(){
    int n;
    cout<<"Enter a Number: ";
    cin>> n;
    int digitSum = sumOfDigits(n);
    if(isPrime(digitSum)){
        cout<<"YES\n";
    }else{
        cout<<"NO\n";
    }

    return 0;
}